### PR TITLE
Add corner detection functionality

### PR DESCRIPTION
**Pull request by yhenon**
This adds corner detection based on the Harris algorithm and the Shi-Tomasi algorithm. These are pretty common image processing tasks, so I thought they may be useful to others.

### CODE REVIEW

File: src/corner.jl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+function imcorner(img::AbstractArray, method::AbstractString="harris", border::AbstractString="replicate", blockSize::Int=3, k::Float64=0.04)
```
Comment (timholy) [inline src/corner.jl]: Should any of these be keyword arguments?

Comment (yhenon) [inline src/corner.jl]: I think that's a good idea, everything but `img` could be a keyword.

**Comment by timholy** [comment]
Need any help with this? Not sure why it was closed. The Travis failures?

**Comment by yhenon** [comment]
Hi Tim,
I was just dealing with a Git issue on my side and was afraid of messing up. I've reopened it now. I've noticed I should add a docstring to the function imcorner() - any other comments ?

**Comment by timholy** [comment]
I totally understand git issues. It was the biggest barrier for me getting started with contributing to julia.

Adding the docstring would be great. Otherwise, everything looks really good!

**Comment by timholy** [comment]
FYI I don't get an email when you update this, so if you make changes and want me to look at it, just post a little comment.

If you click on the red "x" you'll see there's a test failure.

**Comment by yhenon** [comment]
Thanks for all your assistance Tim! I've pushed an update that should fix the test failures.

**Comment by timholy** [comment]
Thanks for the contribution!

**Comment by yhenon** [comment]
Thanks for everything Tim. I hope to be making some more contributions in the fitire.

**Pull request merged**